lib/Dialect/TensorExt/Conversions/TensorExtToTensor/TensorExtToTensor.cpp

```diff
@@ -0,0 +1,118 @@
+#include "lib/Dialect/TensorExt/Conversions/TensorExtToTensor/TensorExtToTensor.h"
+
+#include "lib/Dialect/TensorExt/IR/TensorExtDialect.h"
+#include "lib/Utils/ConversionUtils.h"
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"    // from @llvm-project
+#include "mlir/include/mlir/Dialect/Tensor/IR/Tensor.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+
+namespace mlir::heir::tensor_ext {
+
+#define GEN_PASS_DEF_TENSOREXTTOTENSOR
+#include "lib/Dialect/TensorExt/Conversions/TensorExtToTensor/TensorExtToTensor.h.inc"
+
+class TensorExtToTensorTypeConverter : public TypeConverter {
+ public:
+  TensorExtToTensorTypeConverter(MLIRContext *ctx) {
+    addConversion([](Type type) { return type; });
```

Comment:
If there is no type conversion then we can skip the whole DialectConversion framework and just implement this with the walk-based rewriter.